pkg/globalnet/controllers/ipam/ippool.go

```diff
@@ -118,3 +118,15 @@ func (p *IpPool) RequestIp(key string, ip string) (string, error) {
 	// It is neither allocated for this key, nor available, give another.
 	return p.Allocate(key)
 }
+
+func (p *IpPool) ClearAll() {
```

Comment:
Why do we need this? Won't this be garbage collected?

Reply:
I agree that garbage collector will eventually clean up the memory, but I'm explicitly doing it to be on the safe side instead of leaving it to the Garbage collector. Do you see any harm in this? 

Also, as discussed with Vishal, this cleanup happens on the non-Gateway node, so even if it takes couple of millisecs its not an issue on the non-Gateway node.

Reply:
Clearing maps and nulling out variables does not immediately delete/reclaim the underlying memory - it just makes it eligible to be reclaimed by the GC (whenever it decides to) which, of course, will also occur automatically once all incoming references to the parent objects are released.  There's no harm in manually releasing memory that will be automatically reclaimed anyway but it isn't necessary and won't really cause the memory to be reclaimed any faster.  

A `ClearAll` function would be useful if the caller wants to re-init for reuse.